### PR TITLE
Add ExampleLuxoMuscle as a test

### PR DESCRIPTION
**Pull request by klshrinidhi**
Fixes issue #1676 **partly**.

### Brief summary of changes
* Fix compilation problems with ExampleLuxoMuscle by updating API calls.
* Add ExampleLuxoMuscle as a test.

### CODE REVIEW

File: OpenSim/Tests/ExampleLuxoMuscle/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Define test target and current test dir
+set(TEST_TARGET testExampleLuxoMuscle)
+set(TEST_DIR ${CMAKE_CURRENT_BINARY_DIR})
+
+# Copy dependencies from ExampleMain into the run directory
```
Comment (chrisdembia) [inline OpenSim/Tests/ExampleLuxoMuscle/CMakeLists.txt]: Could you update this comment?

File: OpenSim/Tests/ExampleLuxoMuscle/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    *.obj *.osim *.xml *.sto *.mot)
+
+#
+# Copy example files to run dir
+#
+foreach(dataFile ${TEST_FILES})
+    file(COPY ${dataFile} DESTINATION ${TEST_DIR})
+endforeach(dataFile) 
+
+# Copy the standard answer for example main
```
Comment (chrisdembia) [inline OpenSim/Tests/ExampleLuxoMuscle/CMakeLists.txt]: This comment does not apply for this specific example.

File: OpenSim/Common/STOFileAdapter.h [hunk 3]
```diff
@@ -174,6 +174,9 @@ createSTOFileAdapterForReading(const std::string& fileName);
 std::shared_ptr<DataAdapter>
 createSTOFileAdapterForWriting(const AbstractDataTable& table);
 
+typedef STOFileAdapter_<double> STOFileAdapter;
+typedef STOFileAdapter_<SimTK::Vec3> STOFileAdapterVec3;
```
Comment (chrisdembia) [inline OpenSim/Common/STOFileAdapter.h]: Is this change related to the rest of the PR? I don't see these classes being used in the luxo example.

Comment (chrisdembia) [inline OpenSim/Common/STOFileAdapter.h]: I'm fine with this being included in this PR; I'm just curious how it ended up here.

File: OpenSim/Tests/ExampleLuxoMuscle/CMakeLists.txt (hunk 1 above)
Comment (klshrinidhi) [inline OpenSim/Tests/ExampleLuxoMuscle/CMakeLists.txt]: Done.

File: OpenSim/Tests/ExampleLuxoMuscle/CMakeLists.txt (hunk 2 above)
Comment (klshrinidhi) [inline OpenSim/Tests/ExampleLuxoMuscle/CMakeLists.txt]: Done.

File: OpenSim/Common/STOFileAdapter.h (hunk 3 above)
Comment (klshrinidhi) [inline OpenSim/Common/STOFileAdapter.h]: It is being used in the cpp file.

**Comment by klshrinidhi** [comment]
*Appveyor/PR Win64* failure seems unrelated. Merging.
